content/en/blog/_posts/2021-03-31-kube-state-metrics-goes-v-2-0.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: blog
+title: "kube-state-metrics goes v2.0"
+date: 2021-04-06
+slug: kube-state-metrics-v2-0
+---
+
+**Authors:** Lili Cosic (Red Hat), Frederic Branczyk (Polar Signals), Manuel Rüger (Sony Interactive Entertainment)
+
+## What?
+
+[Kube-state-metrics](https://github.com/kubernetes/kube-state-metrics), a project under the Kubernetes organization, generates Prometheus format metrics based on the current state of the Kubernetes native resources. It does this by listening to the Kubernetes API and gathering information about resources and objects, e.g. Deployments, Pods, Services, and StatefulSets. A full list is available in the [documentation]( https://github.com/kubernetes/kube-state-metrics/tree/master/docs) of kube-state-metrics.
+
+## Why?
+
+After deploying kube-state-metrics it provides you with an array of useful metrics about the state of your cluster out of the box! These metrics can be used to serve as an insight into your cluster: Either through metrics alone, in the form of dashboards, or through an alerting pipeline. They can especially be useful to troubleshoot an outage. A metric worth pointing out is `kube_pod_container_status_restarts_total` which can be used to alert you on a crashing pod. Another helpful metric is `kube_deployment_status_replicas` which together with `kube_deployment_status_replicas_available` can be used to alert on whether a deployment is rolled out successfully or stuck. And there are many more metrics available! If you are interested in further metrics and their details, check out the [documentation](https://github.com/kubernetes/kube-state-metrics/tree/master/docs#readme).
```

Comment:
```suggestion
`kube-state-metrics` provides you with a wide range of useful metrics about the state of your Kubernetes cluster out of the box! These metrics can be used to serve as an insight into your cluster: Either through metrics alone, in the form of dashboards, or through an alerting pipeline. They can especially be useful to troubleshoot an outage. To provide a few examples:
i) `kube_pod_container_status_restarts_total` can be used to alert on a crashing pod.
ii)`kube_deployment_status_replicas` which together with `kube_deployment_status_replicas_available` can be used to alert on whether a deployment is rolled out successfully or stuck. 

And there are many more metrics available at your disposal! To learn more about the other metrics and their details, please check out the [documentation](https://github.com/kubernetes/kube-state-metrics/tree/master/docs#readme).
```

Reply:
I really like that the start of the sentence isn't `kube-state-metrics`, if you can suggest another way, I would prefer to go that way, if that's okay?

Reply:
Maybe ```A wide range of useful metrics about the state of your Kubernetes cluster is provided by `kube-state-metrics` out of the box!```

Reply:
I do like the breakout format of the examples - for the intro piece I and was thinking something along the same lines as @mrueg .
"There are numerous useful metrics and insights provided by `kube-state-metrics` right out of the box!"

Reply:
Done, thanks all! I added also another example as two is not a nice number :D